maestro-studio/web/src/Examples.tsx

```diff
@@ -42,13 +42,24 @@ const Section = ({ deviceScreen, element, title, documentationUrl, codeSnippets
   codeSnippets: string[]
 }) => {
   const toPercent = (n: number, total: number) => `${Math.round((100 * n / total))}%`
+  const elementHasResourceIdIndex = typeof element.resourceIdIndex === 'number'
+  const elementHasTextIndex = typeof element.textIndex === 'number'
   const codeSnippetComponents = codeSnippets.map(codeSnippet => {
     if (codeSnippet.includes('[id]') && !element.resourceId) return null
+    if (codeSnippet.includes('[resource-id-index]') && !elementHasResourceIdIndex) return null
+    if (codeSnippet.includes('[text-index]') && !elementHasTextIndex) return null
     if (codeSnippet.includes('[text]') && !element.text) return null
     if (codeSnippet.includes('[bounds]') && !element.bounds) return null
     if (codeSnippet.includes('tapOn') && isBoundsEmpty(element)) return null
+
+    if (elementHasResourceIdIndex && codeSnippet.includes('[id]') && !codeSnippet.includes('[resource-id-index]')) return null
+    if (elementHasTextIndex && codeSnippet.includes('[text]') && !codeSnippet.includes('[text-index]')) return null
```

Comment:
Observation: this is getting a little hard to follow along -- is there a way we can make this easier to read? Maybe splitting it out into separate functions with descriptive names, adding comments, or something else?

Reply:
Let me add some comments in here